docker/docker/src/main/java/org/arquillian/cube/docker/impl/await/PollingAwaitStrategy.java

```diff
@@ -116,40 +131,66 @@ public boolean await() {
     }
 
     private boolean executeWaitForIt(String containerIp, int port) {
-        if (! new File(WAIT_FOR_IT_SCRIPT_DIRECTORY).exists()) {
-            if (! new File(WAIT_FOR_IT_SCRIPT_DIRECTORY).mkdirs()) {
-                throw new IllegalArgumentException("Couldn't create the Arquillian directory at ~/.arquillian");
+
+        // We copy our wait-for-it-sh.sh file form classpath
+        // Our wait-for-it-sh.sh script also works with busybox/alpine which is not true with the official one.
+
+        try {
+
+            if (!alreadyCopiedWaitForIt) {
+                final Path waitForItLocation = copyWaitForItScriptToTempDir();
+                // Then copy this into the container
+                dockerClientExecutor.copyStreamToContainer(cube.getId(), waitForItLocation.toFile(), new File(CONTAINER_DIRECTORY));
+                alreadyCopiedWaitForIt = true;
             }
-        }
 
-        // Check if in (~/.arquillian/wait-for-it.sh) is the file
-        if (! new File(WAIT_FOR_IT_SCRIPT).exists()) {
-            // If not then download the script
-            Spacelift.task(DownloadTool.class)
-                    .from(WAIT_FOR_IT_SCRIPT_DOWNLOAD_LOCATION)
-                    .to(WAIT_FOR_IT_SCRIPT)
-                    .execute()
-                    .await();
+            String command = resolveWaitForItCommand(containerIp, port);
+            final String[] commands = {"sh", "-c", command};
+            CubeOutput result = dockerClientExecutor.execStart(cube.getId(), commands);
+            return result.getStandard() != null && result.getStandard().trim().contains(MESSAGE);
+        } catch (IOException e) {
+            throw new IllegalArgumentException(e);
         }
-        // Then copy this into the container
-        dockerClientExecutor.copyStreamToContainer(cube.getId(), new File(WAIT_FOR_IT_SCRIPT));
-        // Fix permissions
-        dockerClientExecutor.execStart(cube.getId(), "sh", "-c", "chmod 744 wait-for-it.sh");
-
-        String command = resolveWaitForItCommand(containerIp, port);
-        final String[] commands = {"sh", "-c", command};
-        String result = dockerClientExecutor.execStart(cube.getId(), commands);
-        return result != null && result.trim().contains(MESSAGE);
+
+    }
+
+    private Path copyWaitForItScriptToTempDir() throws IOException {
+        final Path arquilliancube = Files.createTempDirectory("arquilliancube");
+        final Path waitForItLocation = arquilliancube.resolve(Paths.get(WAIT_FOR_IT_SCRIPT));
+        Files.copy(PollingAwaitStrategy.class.getResourceAsStream("/org/arquillian/cube/docker/impl/await/wait-for-it.sh"), waitForItLocation);
```

Comment:
Currently the old one is the one that works. IntelliJ has renamed the previous comment automatically :)